spec/solver_spec.rb

```diff
@@ -0,0 +1,59 @@
+require 'simplecov'
+SimpleCov.start
+require_relative '../solver'
+
+describe 'test a new instance of Solver' do
+  before(:each) do
+    @new_solver = Solver.new
+  end
+
+  context 'test the factorial method' do
+    it 'returns the rigth answer for 5' do
+      expect(@new_solver.factorial(5)).to eq(120)
+    end
+
+    it 'returns the rigth answer for 10' do
+      expect(@new_solver.factorial(10)).to eq(3_628_800)
+    end
```

Comment:
- [ ] Kindly add a test case for a special case. For instance, when the input is 0, the result should equal 1.